Image creation script for Ubuntu 20.04

khabiruddin opened the pull request:
This script will build an image for Ubuntu20.04.

CCCP-3301

khabiruddin committed:
Image creation script for Ubuntu 20.04

This script will build an image for Ubuntu20.04.

CCCP-3301

junousi reviewed (changes requested):
Script looks good. Crontab in same directory needs modifications, too

khabiruddin committed:
Crontab changes for Ubuntu20.04

We going to have Ubuntu20.04 in Our both environment.
Just for now, we will have it in our cPouta environment before ePouta.

CCCP-3301

khabiruddin commented:
> Script looks good. Crontab in same directory needs modifications, too

Yes made crontab changes. This time only for cPouta devel and production before ePouta.

junousi commented on scripts/crontab:
```diff
@@ -12,6 +12,7 @@ SHELL=/bin/bash
 0 6 1,16 * * source $HOME/bin/openrc_cpouta_prod_v3.sh; $HOME/diskimage-builder-scripts.cpouta/scripts/image_create_ubuntu-16.04-cuda.sh >> $HOME/log/c/ubuntu-16.04-cuda.log 2>&1
 
 0 7 1,16 * * source $HOME/bin/openrc_cpouta_prod_v3.sh; $HOME/diskimage-builder-scripts.cpouta/scripts/image_create_ubuntu-18.04.sh >> $HOME/log/c/ubuntu-18.04.log 2>&1
+0 7 1,16 * * source $HOME/bin/openrc_cpouta_prod_v3.sh; $HOME/diskimage-builder-scripts.cpouta/scripts/image_create_ubuntu-20.04.sh >> $HOME/log/c/ubuntu-20.04.log 2>&1
```
You shouldn't do two builds at the same time, it can screw up the loop mount system

khabiruddin replied:
Now each imagebuild at different time.

junousi replied:
```
0 10 * * * [...] >> $HOME/log/c-devel/centos7-cuda.log 2>&1
0 10 1,16 * * [...] >> $HOME/log/c/ubuntu-20.04.log 2>&1
```

These are still overlapping at 10AM on the 1st and 16th day of the month.

I think you need to push one or more of the cpouta-devel builds closer to evening to make this fit.

khabiruddin replied:
Yes made another change for ubuntu-20.04. Now this is 30 10 1,6 * *.

khabiruddin committed:
Not two imagebuilds at same time

We should not run two imagebuilds at the same time.

CCCP-3301

junousi commented:
Also! You could already add the ePouta-prod entry in #commented form

khabiruddin committed:
Different time for each build

Previous crontab overlaps ubuntu-20.04 build with another one.
This time 30mins gaps in between.

CCCP-3301

khabiruddin committed:
Each imagebuild should have one hour interval

Imagebuild should have one hour interval for a safe build.
Within an hour it should have finish one build without starting another
one.

We also have Ubuntu20.04 in ePouta in near future. Commented line
indicates time slot for that.

CCCP-3301

khabiruddin commented:
> Also! You could already add the ePouta-prod entry in #commented form

Yes Ubuntu20.04 build for ePouta in commented for just for now.

khabiruddin commented:
Now I have given 1hour gaps between image build.

junousi commented:
It's a bit strange order not having them in a continuous range, but OTOH it's more outside 9-to-5 office hours or 7-to-19 SLA hours. Works for me.
